src/common/util/deep-equal.ts

```diff
@@ -0,0 +1,106 @@
+// from https://github.com/epoberezkin/fast-deep-equal
+export const equal = (a: any, b: any): boolean => {
```

Comment:
Let's make naming clear since TS autocomplete with a global scope.

```suggestion
export const deepEqual = (a: any, b: any): boolean => {
```